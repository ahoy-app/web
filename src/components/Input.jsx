import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'

import TextBox from './TextBox'
import Spacer from './Spacer'
import RoundButton from './RoundButton'

const styles = {
  // height: '40px',
  width: '100%',

  padding: '5px 10px 10px 10px',
  boxSizing: 'border-box',

  display: 'flex',
}

const Input = ({ onChange, onSend }) => (
  <div style={styles}>
    <TextBox onChange={onChange} onEnter={onSend} />
    <Spacer right={10} />
    <RoundButton onClick={onSend} />
  </div>
)

Input.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSend: PropTypes.func.isRequired,
}

export default Radium(Input)
