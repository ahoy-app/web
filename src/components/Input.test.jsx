import React from 'react'
import { mount } from 'enzyme'

import Input from './Input'

let component

describe('Input', () => {
  afterAll(() => {
    if (component) component.unmount()
  })
  it('should render correctly', () => {
    component = mount(<Input onChange={() => {}} />)
    expect(component).toMatchSnapshot()
  })
})
