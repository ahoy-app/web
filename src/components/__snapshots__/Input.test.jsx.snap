// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<Input
  onChange={[Function]}
  onSend={[Function]}
>
  <div
    data-radium={true}
    style={
      Object {
        "boxSizing": "border-box",
        "display": "flex",
        "padding": "5px 10px 10px 10px",
        "width": "100%",
      }
    }
  >
    <TextBox
      key=".0"
      onChange={[Function]}
    >
      <input
        data-radium={true}
        onChange={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "border": "none",
            "borderRadius": "1rem",
            "boxSizing": "border-box",
            "filter": "drop-shadow(2px 5px 2px #0003)",
            "flex": 1,
            "fontFamily": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
            "fontSize": "1em",
            "height": "40px",
            "lineHeight": "2rem",
            "padding": "0px 10px 0px 10px",
          }
        }
      />
    </TextBox>
    <Spacer
      key=".1"
      right={10}
    >
      <div
        data-radium={true}
        style={
          Object {
            "padding": "0px 10px 0px 0px",
          }
        }
      />
    </Spacer>
    <RoundButton
      key=".2"
      onClick={[Function]}
    >
      <button
        data-radium={true}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "border": "none",
            "borderRadius": "1rem",
            "filter": "drop-shadow(2px 5px 2px #0003)",
            "fontFamily": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
            "fontSize": "1em",
            "height": "40px",
            "lineHeight": "2rem",
            "minWidth": "40px",
          }
        }
      />
    </RoundButton>
  </div>
</Input>
`;
